### PR TITLE
feat: make bindings, wallet and txManager available

### LINKED ISSUE #220: Create a BuildAll constructor for low-level interactions
currently the clients.BuildAll() function returns high-level chainio clients.
Power users might want direct access to bindings + wallet/txmgr, so we should also make those available

Should prob also write improve the chainio README to show these 2 different ways of interacting with contracts (high-level and low-level).

### PR DESCRIPTION
**Pull request by renlulu**
Fixes #220 .

### Reviewer Checklist
- [ ] Code is well-documented
- [ ] Code adheres to Go [naming conventions](https://go.dev/doc/effective_go#names)
- [ ] Code deprecates any old functionality before removing it

### CODE REVIEW

**Comment by renlulu** [comment]
I will rebase after #203 get merged, the diff will be much more less by then.

File: chainio/clients/builder.go [hunk 1]
```diff
@@ -36,15 +37,19 @@ type BuildAllConfig struct {
...
+	AvsRegistryChainSubscriber  *avsregistry.AvsRegistryChainSubscriber
+	AvsRegistryChainWriter      *avsregistry.AvsRegistryChainWriter
+	ElChainReader               *elcontracts.ELChainReader
+	ElChainWriter               *elcontracts.ELChainWriter
+	EthHttpClient               eth.Client
+	EthWsClient                 eth.Client
+	Wallet                      wallet.Wallet
+	TxManager                   txmgr.TxManager
+	AvsRegistryContractBindings *utils.AvsRegistryContractBindings
+	EigenlayerContractBindings  *utils.EigenlayerContractBindings
```
Comment (samlaf) [inline chainio/clients/builder.go:50]: @shrimalmadhur any oppositions to returning everything but the kitchen sink in this constructor?
I was thinking we break it up into smaller builders (for eg one for high-level chainio reader/writer/subscribers, and another for low-level interactions), but I don't see any problem with returning everything in here and letting user take whatever they need?

Comment (renlulu) [reply chainio/clients/builder.go:50]: let me know if you prefer break it, i will adjust my PR, thanks!

Comment (shrimalmadhur) [reply chainio/clients/builder.go:50]: IMO, 
we should have 3 exported Methods
`BuildELClients` - just returns core EL contract Bindings
`BuildAVSRegistryClients` - just returns AVS resgistry bindings
`BuildAllClients` - returns all
@samlaf how about this and then user can use based on whatever is needed? I don't want someone to force get AVS registry if not needed.

File: chainio/clients/builder.go [hunk 2]
```diff
@@ -180,7 +218,7 @@ func (config *BuildAllConfig) buildElClients(
 		txMgr,
 	)
 
-	return elChainReader, elChainWriter, nil
+	return elChainReader, elChainWriter, avsRegistryContractBindings, elContractBindings, nil
```
Comment (shrimalmadhur) [inline chainio/clients/builder.go:221]: wait why is this retuning `avsRegistryContractBindings` even tho it's not returning any AVS clients?

Comment (samlaf) [reply chainio/clients/builder.go:221]: true this shouldn't be here

Comment (renlulu) [reply chainio/clients/builder.go:221]: Okie, i am adjusting my PR.

Comment (samlaf) [reply chainio/clients/builder.go]: lgtm now, thanks

File: chainio/clients/builder.go (hunk 1 above)
Comment (samlaf) [reply chainio/clients/builder.go:50]: https://github.com/Layr-Labs/eigensdk-go/pull/117 this PR does this. I think we can merge that one first then and rebase this one on top?

Comment (shrimalmadhur) [reply chainio/clients/builder.go:50]: yes +1 to that.

Comment (renlulu) [reply chainio/clients/builder.go]: Oh really sorry, i didn't notice that there is a PR already! Just make my PR more clean...

Comment (renlulu) [reply chainio/clients/builder.go]: I am totally okay you can go with that one first!

Comment (samlaf) [reply chainio/clients/builder.go]: @renlulu no worries, that was an old one... just merged the other PR. Can you rebase here?

Comment (renlulu) [reply chainio/clients/builder.go]: GM, just rebased!

**Review by samlaf — APPROVED** [review]
This works for now. Don't think it solves what I had in mind for #220 so I will leave that open. But at least we have this for now. Thanks :)

**Pull request merged**